### PR TITLE
Bring back local create-expo-module templates and fix require error.

### PR DESCRIPTION
**Pull request by aleqsio**
# Why

@brent reverted https://github.com/expo/expo/commit/ac6609f16bf8e7824e558071ae65cc3993dc2025 in https://github.com/expo/expo/commit/bac067c60dd2660b11deed4e73b655c6eb589a0e due to crashes.
We want to land it back on main.

# How

`require` was looking for the expo package version in `.npx` instead of in the project directory.

I'm fixing this by using `require.resolve` and passing cwd as a paths directory. 

# Test Plan

I've now made sure that this is using the package version from the actual project, not from the monorepo.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/create-expo-module/src/create-expo-module.ts [hunk 1]
```diff
@@ -210,14 +210,41 @@ async function getNpmTarballUrl(packageName: string, version: string = 'latest')
   return stdout.trim();
 }
 
+/**
+ * Gets expo SDK version major from the local package.json.
+ */
+async function getLocalSdkMajorVersion(): Promise<string | null> {
+  const path = require.resolve('expo/package.json', { paths: [process.cwd()] });
+  if (!path) return null;
```
Comment (tsapeta) [inline packages/create-expo-module/src/create-expo-module.ts:218]: Please, stop doing that 😛 
```suggestion
  if (!path) {
    return null;
  }
```

File: packages/create-expo-module/src/create-expo-module.ts [hunk 2]
```diff
@@ -210,14 +210,41 @@ async function getNpmTarballUrl(packageName: string, version: string = 'latest')
...
+  const { version } = require(path) ?? {};
```
Comment (tsapeta) [inline packages/create-expo-module/src/create-expo-module.ts:219]: What do you think if we wrap it in try/catch and handle it nicely? Maybe an appropriate console.error, but with a fallback to `latest` anyway?

File: packages/create-expo-module/src/create-expo-module.ts [hunk 3]
```diff
@@ -210,14 +210,41 @@ async function getNpmTarballUrl(packageName: string, version: string = 'latest')
...
+  return version?.split('.')[0] ?? null;
+}
+
+/**
+ * Selects correct version of the template based on the SDK version for local modules and EXPO_BETA flag.
+ */
+async function getTemplateVersion(targetDir: string, isLocal) {
```
Comment (tsapeta) [inline packages/create-expo-module/src/create-expo-module.ts:226]: Interesting that linter doesn't catch this 🤔 
```suggestion
async function getTemplateVersion(isLocal: boolean) {
```

**Review by tsapeta — APPROVED** [review]

File: packages/create-expo-module/src/create-expo-module.ts (hunk 1 above)
Comment (aleqsio) [reply packages/create-expo-module/src/create-expo-module.ts:218]: sorry 🥺

File: packages/create-expo-module/src/create-expo-module.ts (hunk 3 above)
Comment (aleqsio) [reply packages/create-expo-module/src/create-expo-module.ts]: Apparently it's a TS message, but not a build error, but the linter doesn't pick it up
```
'targetDir' is declared but its value is never read.ts(6133)
```

File: packages/create-expo-module/src/create-expo-module.ts (hunk 2 above)
Comment (aleqsio) [reply packages/create-expo-module/src/create-expo-module.ts:221]: I've added a try around calling this function

**Review by tsapeta — APPROVED** [review]

**Pull request merged**